frappe/desk/calendar.py

```diff
@@ -44,6 +44,8 @@ def get_events(doctype, start, end, field_map, filters=None, fields=None):
 		fields = [field_map.start, field_map.end, field_map.title, 'name']
 
 	if field_map.color:
+		if isinstance(fields, str):
+			fields = json.loads ( fields )
```

Comment:
```suggestion
		fields = frappe.parse_json(fields)
```
`frappe.parse_json` handles this. Also can you move this piece of code to the top of the function?

Reply:
Hi made those changes, worked fine. Can you help me with #12046. Also I have submitted some UX issues you guys should really fix. If you want to compete with Netsuite and SAP you must have flawless UX. When I did demo to key decision makers they did not see the great engine you have built, they only see the UX issues I raised and reject Frappe/ErpNext as "a-bit-broken"